engine/src/main/java/graphql/nadel/engine/execution/ServiceResultNodesToOverallResult.java

```diff
@@ -594,12 +590,46 @@ public TraversalControl enter(TraverserContext<ExecutionResultNode> context) {
         });
     }
 
-    private List<String> getFieldIdsWithoutTransformationId(ExecutionResultNode node, TransformationMetadata transformationMetadata) {
-        return node.getFieldIds().stream().filter(fieldId -> FieldMetadataUtil.getTransformationIds(fieldId, transformationMetadata.getMetadataByFieldId()).size() == 0).collect(Collectors.toList());
+    private ExecutionResultNode nodesWithoutTransformationIds(ExecutionResultNode executionResultNode, Set<String> transformationIds, TransformationMetadata transformationMetadata) {
+
+        return resultNodesTransformer.transform(executionResultNode, new TraverserVisitorStub<ExecutionResultNode>() {
+
+            @Override
+            public TraversalControl enter(TraverserContext<ExecutionResultNode> context) {
+                ExecutionResultNode node = context.thisNode();
+                List<String> fieldIdsWithId;
```

Comment:
Does `getFieldIdsWithoutTransformationIds` return a list of fieldIds with or without ids? I'm just checking that the name of this var is not in conflict with what the method is actually returning. It seems to me that the method will return a list of fields **without** (transformation) ids?

Reply:
yeah the naming is wrong here.